docs/DeveloperGuide.md

```diff
@@ -158,103 +158,260 @@ Classes used by multiple components are in the `seedu.address.commons` package.
+1. The user inputs the command `delete 1`, intending to delete a person with index 1 in the contact list.
+2. The `DeleteCommandParser` interprets the input.
+3. A `DeleteCommand` object is created.
+4. The `LogicManager` invokes the execute method of DeleteCommand.
+5. The execute method of `DeleteCommand` invokes the `deletePerson` method in `Model` property to delete the specified `Person` object.
+6. The execute method of `DeleteCommand` returns a `CommandResult` object which stores the data regarding the completion of the `DeleteCommand`.
+7. The UI reflects this new list without the deleted `Person`.
 
-<puml src="diagrams/UndoSequenceDiagram-Model.puml" alt="UndoSequenceDiagram-Model" />
+<div markdown="span" class="alert alert-info">:information_source: **Note**:
 
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+- At step 2, if input is detected as invalid, an error will be shown on the screen and the `DeleteCommand` object will NOT be created!
+</div>
 
-<box type="info" seamless>
+---
+### Edit an Udder Feature
 
-**Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+#### **Feature**
 
-</box>
+`edit INDEX [n/NAME] [p/PHONE NUMBER] [e/EMAIL ADDRESS] [a/ADDRESS] [r/ROLE] [m/MAJOR] [t/TAG]`,
+where all the fields are optional.
+
+#### **Feature Purpose**
+
+The `edit` command allows users to edit the details of an existing `Person` in the `AddressBook`. 
+
+#### Key Components
+- `EditCommand`: Executes the action of editing an existing Udder based on user input. This command processes the input parameters to update the Person object's details within FindingbrUdders.
+- `EditCommandParser`: Parses user input to create an `EditCommand` object.
+- `LogicManager`: Invokes the `EditCommand` to execute the editing of an Udder.
+- `ModelManager`: Implements the `Model` interface and contains the internal list of persons (Udders).
+- `Person`: Represents an Udder in FindingbrUdders, encapsulating their personal information.
+- `AddressBookParser`: Creates an `EditCommand` object based on the user input.
+
+### **Sequence of action**
+
+To help you understand how the `edit` command works, here is a list of steps illustrating what occurs when [`LogicManager#execute()` is invoked](#logic-component):
+
+We will be using the user input `edit 1 n/John Doe p/98765432 e/johnd@example.com a/123 Main St r/brUdder m/bza` as an example, whereby the original `Person` object has an address of `John street, block 123, #01-01`.
```

Comment:
Is it due to the fact that we did not change the package name, the class is still called `Person` instead of `Udder`?